cmd/virt-launcher/virt-launcher.go

```diff
@@ -302,6 +302,26 @@ func waitForFinalNotify(deleteNotificationSent chan watch.Event,
 	}
 }
 
+// writeProtectPrivateDir waits until the kubevirt private vnc socket exists and than mark its folder as read only
+// this is a workaround preventing QEMU from deleting its sockets prematurely as described in a bug https://bugs.launchpad.net/qemu/+bug/1795100
+// once the QEMU 4.0 is released the need for this workaround goes away
+// Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1683964
+func writeProtectPrivateDir(uid string) {
```

Comment:
would it make sense to move this into pkg/virt-launcher/monitor.go, and call it from InitializePrivateDirectories()? There the socket is created, and we have the path already available.

Reply:
This is a good point, I have originally considered monitor.go and wait in the  InitializePrivateDirectories. The reason why the call is here is I require the VM to be started and QEMU to report it back. This makes the wait for socket creation as short as possible.

The function implementation can be safely placed here, as I will be removing it once the QEMU 4.0 is released, which should be soon. RC2 is already available.

Would this be ok?

Reply:
ack, works for me